assets/JS/index.js

```diff
@@ -13,16 +13,27 @@ $(document).ready(function () {
         .then(function (data) {
             console.log(data);
             const categoryArray = data.trivia_categories;
-            $.each(categoryArray, function (i) {
+            categoryArray.forEach(function(category) {
                 categoryInput.append(
-                    $('<option></option>').val(categoryArray[i].id).html(categoryArray[i].name)
+                  $('<option></option>').val(category.id).html(category.name)
                 );
             });
             $('select').formSelect();
-        });
-        submitButton.addEventListener("click", submitModalForm);
-        
-});
+        })
+        .catch(function (error) {
+          console.error('Error fetching categories', error)
+        })
+        categoryInput.on('change', function () {
```

Comment:
Pls get rid of this **categoryInput.on**. I'll handle this in **submitModalForm**. We don't need to override localstorage every time when category change, only on **form submit**, since this will be user "final decision"